### PR TITLE
[Unity] `nn.Mutator` for `nn.Module` level transform

### PR DESCRIPTION
**Pull request by cyx-6**
This PR introduces the `nn.Mutator` for the `nn.Module` level transform.

cc: @junrushao

### CODE REVIEW

File: python/tvm/relax/frontend/nn/core.py [hunk 1]
```diff
@@ -520,6 +520,87 @@ def forward(self, x):  # pylint: disable=invalid-name
...
+        node : Any
+            The current node to visit.
+
+        Returns
+        ------
+        ret_node: Any
+            The new node to replace current node.
+        """
+        for key, value in node.__dict__.items():
+            if isinstance(value, Module):
```
Comment (junrushao) [inline python/tvm/relax/frontend/nn/core.py:595]: You might want to check ModuleList as well, which may need some special handling

File: tests/python/relax/test_frontend_nn_mutator.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Any
+
+import tvm
+from tvm.relax.frontend import nn
+
+
+def test_mutator_naming():
```
Comment (junrushao) [inline tests/python/relax/test_frontend_nn_mutator.py:24]: Add a test for ModuleList

**Review by junrushao — COMMENTED** [review]
Overall it looks great!

File: python/tvm/relax/frontend/nn/core.py [hunk 3]
```diff
@@ -520,6 +520,87 @@ def forward(self, x):  # pylint: disable=invalid-name
         return x
 
 
+class Matuator:
```
Comment (junrushao) [inline python/tvm/relax/frontend/nn/core.py:523]: move this to `visitor.py`

File: tests/python/relax/test_frontend_nn_mutator.py (hunk 2 above)
Comment (cyx-6) [reply tests/python/relax/test_frontend_nn_mutator.py]: added

File: python/tvm/relax/frontend/nn/core.py (hunk 3 above)
Comment (cyx-6) [reply python/tvm/relax/frontend/nn/core.py]: moved

**Review by junrushao — APPROVED** [review]

**Pull request merged**